src/dataaggregator/policy/policy_test.go

```diff
@@ -0,0 +1,120 @@
+package policy_test
+
+import (
+	. "dataaggregator/policy"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Policy", func() {
+
+	var testAppId = "testAppId"
+	var testAppIdAnother = "testAppIdAnother"
+	var trigger *Trigger
+	var policyStr = `
+   {
+   "instance_min_count":1,
+   "instance_max_count":5,
+   "scaling_rules":[
+      {
+         "metric_type":"MemoryUsage",
+         "stat_window_secs":300,
+         "breach_duration_secs":300,
+         "threshold":30,
+         "operator":"<",
+         "cool_down_secs":300,
+         "adjustment":"-1"
+      }
+   ]
+}`
+	var policyStrAnother = `
+   {
+   "instance_min_count":2,
+   "instance_max_count":5,
+   "scaling_rules":[
+      {
+         "metric_type":"MemoryUsage",
+         "stat_window_secs":300,
+         "breach_duration_secs":600,
+         "threshold":30,
+         "operator":"<",
+         "cool_down_secs":300,
+         "adjustment":"-1"
+      }
+   ]
+}`
+	var p1, p2, policyJson *PolicyJson
+	Context("PolicyJson.Equals", func() {
+		Context("when p1 and p2 are all nil", func() {
+			BeforeEach(func() {
+				p1 = nil
+				p2 = nil
+			})
+			It("should return true", func() {
+				Expect(p1.Equals(p2)).To(Equal(true))
+			})
+		})
+		Context("when only one of p1 and p2 are is nil", func() {
+			BeforeEach(func() {
+				p1 = nil
+				p2 = &PolicyJson{AppId: testAppId, PolicyStr: policyStr}
+			})
+			It("should return false", func() {
+				Expect(p1.Equals(p2)).To(Equal(false))
+			})
+		})
+		Context("when p1 and p2 are both not nil", func() {
+			Context("when the AppId and PolicyStr are both the same", func() {
+				BeforeEach(func() {
+					p1 = &PolicyJson{AppId: testAppId, PolicyStr: policyStr}
+					p2 = &PolicyJson{AppId: testAppId, PolicyStr: policyStr}
+				})
+				It("should return true", func() {
+					Expect(p1.Equals(p2)).To(Equal(true))
+				})
+			})
+			Context("when the AppId are the same ,while the  PolicyStr are different", func() {
+				BeforeEach(func() {
+					p1 = &PolicyJson{AppId: testAppId, PolicyStr: policyStr}
+					p2 = &PolicyJson{AppId: testAppId, PolicyStr: policyStrAnother}
+				})
+				It("should return false", func() {
+					Expect(p1.Equals(p2)).To(Equal(false))
+				})
+			})
+			Context("when the AppId are different ,while the  PolicyStr are the same", func() {
+				BeforeEach(func() {
+					p1 = &PolicyJson{AppId: testAppId, PolicyStr: policyStr}
+					p2 = &PolicyJson{AppId: testAppIdAnother, PolicyStr: policyStr}
+				})
+				It("should return false", func() {
+					Expect(p1.Equals(p2)).To(Equal(false))
+				})
+			})
+
+		})
+	})
+	Context("GetTriggerFromPolicy", func() {
+
+		BeforeEach(func() {
+			policyJson = &PolicyJson{AppId: testAppId, PolicyStr: policyStr}
+			trigger = policyJson.GetTrigger()
+		})
+		It("should return a trigger", func() {
+			Expect(trigger.AppId).To(Equal(testAppId))
```

Comment:
You could just do a Equals(&Trigger {
....
})
rather that each line which could overlook new fields.